arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md

```diff
@@ -0,0 +1,150 @@
+---
+title: "How to customize your Orbit chain's precompiles"
+sidebar_label: "Customize your chain's precompiles"
+description: "Learn how (and when) to customize your Orbit chain's precompiles"
+author: jasonwan
+sidebar_position: 2
+---
+
+import PublicPreviewBannerPartial from '../partials/_orbit-public-preview-banner-partial.md';
+
+<PublicPreviewBannerPartial />
+
+import UnderConstructionPartial from '../../partials/_under-construction-banner-partial.md';
+
+<UnderConstructionPartial />
+
+There are two ways to customize your chain's precompiles:
+
+1.  Add new methods to to an existing [precompile](https://github.com/OffchainLabs/nitro-contracts/tree/97cfbe00ff0eea4d7f5f5f3afb01598c19ddabc4/src/precompiles).
+2.  Create a new precompile.
+
+### Prerequisites
+
+Clone the Nitro repository before you begin:
+
+```shell
+git clone <https://github.com/OffchainLabs/nitro.git>
+cd nitro
+git submodule update --init --recursive --force
+```
+
+## Option 1: Add new methods to an existing precompile
+
+Using your favorite code editor, open an existing precompile from the [precompiles/](https://github.com/OffchainLabs/nitro/tree/master/precompiles) directory. We'll use `ArbSys.sol` as an example. Open the the corresponding `go` file (`ArbSys.go`) and add a simple `SayHi` method:
```

Comment:
```suggestion
Using your favorite code editor, open the implementation file of an existing precompile from the [precompiles/](https://github.com/OffchainLabs/nitro/tree/master/precompiles) directory. We'll use `ArbSys.sol` as an example. Open the the corresponding `go` file (`ArbSys.go`) and add a simple `SayHi` method:
```

In the precompiles reference page, we talk about the "go implementation file" and the "solidity interface file" to make a distinction between both assets, so it is more clear for the user. I've added one potential example if you want to use the same language.

Reply:
Love this - thank you